packages/smooth_app/lib/pages/product_list_user_dialog_helper.dart

```diff
@@ -403,44 +331,76 @@ class _UserLists extends StatelessWidget {
             title: Text(name),
             onTap: () {
               if (selectedLists.contains(name)) {
-                onListUnselected(name);
+                widget.onListUnselected(name);
+                selectedLists.removeWhere((String e) => e == name);
               } else {
-                onListSelected(name);
+                widget.onListSelected(name);
+                selectedLists.add(name);
               }
+              setState(() {});
             });
       }).toList(growable: false),
     );
   }
 }
 
-class _UserEmptyLists extends StatelessWidget {
-  const _UserEmptyLists({Key? key}) : super(key: key);
+/// Widget indicate that the user has no lists yet
+class _UserEmptyLists extends StatefulWidget {
```

Comment:
I'm a bit surprised that you need a special widget when there are no user lists yet - displaying "No list yet" would make sense, but probably in the same dialog. What happens when you add a list: does that close the current dialog and opens another one with a one-item list?
At least please add some `///` comments.

Reply:
Thats a good concern. That was my plan with the pop'ed true and false. In my next commit it now opens the correct alertDialog when a new list is added